integration-tests/tests/build.gradle

```diff
@@ -11,3 +13,31 @@ dependencies {
 
   kaptTest deps.dagger2.compiler
 }
+
+// Okay, this check is sketchy here. But this module serves as integration test so it's not too
+// terrible. What would be worse is setting up integration tests in the Gradle plugin itself,
+// because it requires so much code and might not represent the real world usage.
+//
+// Kotlin 1.3 is broken. The hacky workarounds don't always work. The issue seems to be gone with
+// Kotlin 1.4, so enable this check only for the newer version.
+
+// TODO: remove this check when upgrading to Kotlin 1.4.
```

Comment:
nit: any GitHub issue that this TODO can be tied to?

Reply:
The PR is already up for that. I'll rebase the other PR after I merged this one and remove the TODO again.